tests/purs/failing/OverlappingUnnamedInstances.purs

```diff
@@ -0,0 +1,17 @@
+-- @shouldFailWith OverlappingInstances
+module Main where
+
+class Test a where
+  test :: a -> a
+
+instance Test a where
+  test x = x
+
+instance Test Int where
+  test _ = 0
+
+-- The OverlappingInstances instances error only arises when there are two
+-- choices for a dictionary, not when the instances are defined. So without
+-- `value` this module would not raise a warning.
```

Comment:
From https://github.com/purescript/purescript/pull/4099#discussion_r644345398:
```suggestion
-- `value` this module would not raise a error.
```

Reply:
Hmm, nice catch. Guess this would apply to the test of which this is a copy, too.